tests/integration/api_v2/http/transactions.test.js

```diff
@@ -0,0 +1,261 @@
+				.toMap(transactionSchemaVersion5, { moduleAssetName: refTransaction.moduleAssetName }));
+			expect(response.meta).toMap(metaSchema);
+		});
+
+		it('invalid moduleAssetId -> 400', async () => {
+			const response = await api.get(`${endpoint}?moduleAssetId=-124`, 400);
+			expect(response).toMap(badRequestSchema);
+		});
+
+		it('empty moduleAssetId ->  ok', async () => {
+			const response = await api.get(`${endpoint}?moduleAssetId=`);
+			expect(response).toMap(goodRequestSchema);
+			// expect(response.data).toBeArrayOfSize(10);
+			response.data.forEach(transaction => expect(transaction).toMap(transactionSchemaVersion5));
+			expect(response.meta).toMap(metaSchema);
+		});
+	});
+
+	describe('Retrieve a transaction by transaction ID', () => {
+		it('returns requested transaction with known transaction id', async () => {
+			const response = await api.get(`${endpoint}?id=${refTransaction.id}`);
+			expect(response).toMap(goodRequestSchema);
+			expect(response.data).toBeArrayOfSize(1);
+			response.data.forEach(transaction => expect(transaction)
+				.toMap(transactionSchemaVersion5, { id: refTransaction.id }));
+			expect(response.meta).toMap(metaSchema);
+		});
+
+		it('long transaction id -> 500', async () => {
+			const response = await api.get(`${endpoint}?id=412875216073141752800000`, 500);
+			expect(response).toMap(badRequestSchema);
+		});
+
+		it('invalid transaction id -> 500', async () => {
+			const response = await api.get(`${endpoint}?id=41287`, 500);
+			expect(response).toMap(notFoundSchema);
+		});
+	});
+
+	describe('Retrieve transaction list by block ID', () => {
+		it('known block -> ok', async () => {
+			const response = await api.get(`${endpoint}?block=${refTransaction.block.id}`);
+			expect(response).toMap(goodRequestSchema);
+			expect(response.data).toBeInstanceOf(Array);
+			expect(response.data.length).toBeGreaterThanOrEqual(1);
+			response.data.forEach(transaction => expect(transaction)
+				.toMap(transactionSchemaVersion5, { blockId: refTransaction.blockId }));
+			expect(response.meta).toMap(metaSchema);
+		});
+
+		it('invalid block -> 404', async () => {
+			const response = await api.get(`${endpoint}?block=1000000000000000000000000'`, 404);
+			expect(response).toMap(badRequestSchema);
+		});
+
+		it('empty block param -> ok', async () => {
+			const response = await api.get(`${endpoint}?block=`);
+			expect(response).toMap(goodRequestSchema);
+			expect(response.data).toBeArrayOfSize(10);
+			response.data.forEach(transaction => expect(transaction).toMap(transactionSchemaVersion5));
+			expect(response.meta).toMap(metaSchema);
+		});
+	});
+
+	describe('Retrieve transaction list by height', () => {
+		it('known height -> ok', async () => {
+			const response = await api.get(`${endpoint}?height=${refTransaction.block.height}`);
+			expect(response).toMap(goodRequestSchema);
+			expect(response.data).toBeInstanceOf(Array);
+			expect(response.data.length).toBeGreaterThanOrEqual(1);
```

Comment:
Please also add upper limit checks whereever missing.
```suggestion
			expect(response.data.length).toBeGreaterThanOrEqual(1);
			expect(response.data.length).toBeLessThanOrEqual(10);
```